src/main/scala/io/iohk/ethereum/blockchain/sync/SyncStateSchedulerActor.scala

```diff
@@ -2,33 +2,80 @@ package io.iohk.ethereum.blockchain.sync
 
 import akka.actor.{Actor, ActorLogging, ActorRef, Props, Timers}
 import akka.util.ByteString
+import cats.data.NonEmptyList
 import io.iohk.ethereum.blockchain.sync.LoadableBloomFilter.BloomFilterLoadingResult
-import io.iohk.ethereum.blockchain.sync.SyncStateDownloaderActor.{CancelDownload, RegisterScheduler}
-import io.iohk.ethereum.blockchain.sync.SyncStateScheduler.{ProcessingStatistics, SchedulerState, SyncResponse}
-import io.iohk.ethereum.blockchain.sync.SyncStateSchedulerActor.{
-  BloomFilterResult,
-  GetMissingNodes,
-  MissingNodes,
-  PrintInfo,
-  PrintInfoKey,
-  RestartRequested,
-  StartSyncingTo,
-  StateSyncFinished,
-  SyncStateSchedulerActorCommand,
-  StateSyncStats,
-  WaitingForNewTargetBlock
+import io.iohk.ethereum.blockchain.sync.PeerRequestHandler.ResponseReceived
+import io.iohk.ethereum.blockchain.sync.SyncStateScheduler.{
+  CriticalError,
+  ProcessingStatistics,
+  SchedulerState,
+  SyncResponse
 }
+import io.iohk.ethereum.blockchain.sync.SyncStateSchedulerActor._
+import io.iohk.ethereum.crypto.kec256
+import io.iohk.ethereum.network.p2p.messages.PV63.{GetNodeData, NodeData}
+import io.iohk.ethereum.network.{Peer, PeerId}
 import io.iohk.ethereum.utils.ByteStringUtils
 import io.iohk.ethereum.utils.Config.SyncConfig
 import monix.execution.Scheduler
 
+import scala.annotation.tailrec
+import scala.collection.immutable.Queue
+import scala.concurrent.Future
 import scala.concurrent.duration._
 
-class SyncStateSchedulerActor(downloader: ActorRef, sync: SyncStateScheduler, syncConfig: SyncConfig)
-    extends Actor
+class SyncStateSchedulerActor(
+    sync: SyncStateScheduler,
+    val syncConfig: SyncConfig,
+    val etcPeerManager: ActorRef,
+    val peerEventBus: ActorRef,
+    val scheduler: akka.actor.Scheduler
```

Comment:
I would rather not do that, as this name is expected in both traits i.e `BlacklistSupport` and `PeerListSupport`, and changing it will probably introduce a lot of noise into pr.